components/style/values/generics/grid.rs

```diff
@@ -354,8 +354,13 @@ pub enum RepeatCount {
 
 impl Parse for RepeatCount {
     fn parse<'i, 't>(context: &ParserContext, input: &mut Parser<'i, 't>) -> Result<Self, ParseError<'i>> {
-        if let Ok(i) = input.try(|i| Integer::parse(context, i)) {
+        // Maximum number of repeat is 10000. The greater numbers should be clamped.
+        const MAX_LINE: i32 = 10000;
+        if let Ok(mut i) = input.try(|i| Integer::parse(context, i)) {
             if i.value() > 0 {
+                if i.value() > MAX_LINE {
```

Comment:
This is where we should clamp actually. Because we are converting the `repeat(3, 20px)` function into `20px 20px 20px` in parsing side(thanks to gecko specified value behavior). So we have to clamp here. Otherwise we are getting an OOM crash in [this test](http://searchfox.org/mozilla-central/rev/a83a4b68974aecaaacdf25145420e0fe97b7aa22/layout/style/test/test_grid_container_shorthands.html#171).